src/utils/tracked-controls.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Return enumerated gamepads matching id prefix.
+ *
+ * @param {object} idPrefix - prefix to match in gamepad id, if any.
+ */
+module.exports.getGamepadsByPrefix = function (idPrefix) {
+  var rtn = [];
+  var gamepads = navigator.getGamepads && navigator.getGamepads();
+  if (gamepads) {
+    for (var i = 0; i < gamepads.length; ++i) {
+      var gamepad = gamepads[i];
+      if (gamepad) {
+        if (!idPrefix || idPrefix === '' || gamepad.id.indexOf(idPrefix) === 0) {
+          rtn.push(gamepad);
+        }
+      }
+    }
+  }
+  return rtn;
+};
+
+/**
+ * Enumerate controllers (as built by system tick, e.g. that have pose) and check if they match parameters.
+ *
+ * @param {object} idPrefix - prefix to match in gamepad id, if any.
+ * @param {object} queryObject - map of values to match (hand; index among controllers with idPrefix)
+ */
+module.exports.isControllerPresent = function (idPrefix, queryObject) {
+  var isMatch = false;
+  var sceneEl = document.querySelector('a-scene');
+  var gamepads = sceneEl && sceneEl.systems['tracked-controls'] && sceneEl.systems['tracked-controls'].controllers;
+  if ((!gamepads || gamepads.length === 0) && sceneEl && sceneEl.systems['tracked-controls']) {
+    sceneEl.systems['tracked-controls'].rebuildControllerList();
+    gamepads = sceneEl.systems['tracked-controls'].controllers;
```

Comment:
not entirely... the extra bit at the end is to prevent failures if no scene element yet, or no tracked-controls system injected yet.  but i will refactor to avoid the objection